crates/ty_server/src/server/api.rs

```diff
@@ -352,7 +342,13 @@ fn sync_notification_task<N: traits::SyncNotificationHandler>(
         if let Err(err) = N::run(session, client, params) {
             tracing::error!("An error occurred while running {id}: {err}");
             client.show_error_message("ty encountered a problem. Check the logs for more details.");
+
+            return;
         }
+
+        // if there's any pending workspace diagnostic long-polling request.
+        // Continue it if the session revision changed (e.g. because some document changed).
```

Comment:
```suggestion
        // If there's any pending workspace diagnostic long-polling request,
        // resume it if the session revision changed (e.g. because some document changed).
```